docs/docs/reference/handlers/postgres.md

```diff
@@ -7,26 +7,136 @@ permalink: docs/reference/handlers/postgres.html
 ---
 
 ## PostgreSQL
-The PostgreSQL handler authenticates incoming connections for a particular
-listener.
 
-**Handler in alpha - not suitable for production** 
+**WARNING: Handler in alpha - not suitable for production** 
 
-### Credentials
-- `address`  
-_Required_  
-Connection string of the form `host[:port][/dbname]`  
+The PostgreSQL handler authenticates and brokers connections to a postgres
+database.
 
-- `username`  
-_Required_  
-Username of the PostgreSQL account to connect as  
+To secure connections, we support all the postgres SSL options you're familar
+with. See the `sslmode` option below for details.
 
-- `password`  
-_Required_  
-Password of the PostgreSQL account to connect with  
+Note that, unlike most clients, the default `sslmode` for Secretless is
+`required`, since nearly all use cases require TLS.  If you do need to turn it
+off, however, and know you can do so safely, you can. 
+
+### Configuring the Handler
+
+You tell Secretless where to find your database connection details in the yaml
+file's `credentials` section.
+
+There you specify where to find your database's address, your username and password, 
+as well as the `sslmode` details, including the location of any relevant certificates
+and revocation lists, if applicable.
+
+The options are as follows:
+
+- `address`
+  _Required_
+
+  Connection string of the form `host[:port][/dbname]`  
+
+- `username`
+  _Required_
+
+  Username of the PostgreSQL account you are connecting as.
+
+- `password`
+  _Required_
+
+  Password of the PostgreSQL account you are connecting as.
+
+- `sslmode`
+  _Optional_
+
+  This option determines if the connection between Secretless and your database
+  will be protected by SSL.
+  
+  NOTE: As mentioned above, the default is `require` as opposed to `prefer`,
+  forcing SSL unless you explicitly turn it off.
+
+  The Postgres documentation website provides detail on the [levels of protection](https://www.postgresql.org/docs/9.1/libpq-ssl.html#LIBPQ-SSL-PROTECTION)
+  provided by different values for the sslmode parameter.
+
+  There are [six modes](https://www.postgresql.org/docs/9.1/libpq-connect.html#LIBPQ-CONNECT-SSLMODE):
+
+  + `disable`
+  only try a non-SSL connection
+
+  + `allow`
+  first try a non-SSL connection; if that fails, try an SSL connection
+
+  + `prefer`
+  first try an SSL connection; if that fails, try a non-SSL connection
+
+  + `require` (default)
+  only try an SSL connection. If a root CA file is present, verify the certificate in the same way as if verify-ca was specified
+
+  + `verify-ca`
+  only try an SSL connection, and verify that the server certificate is issued by a trusted certificate authority (CA).
+
+  + `verify-full`
+  only try an SSL connection, verify that the server certificate is issued by a trusted CA and that the server host name matches that in the certificate
+
+_NOTE_: If `sslmode` is set to `require`, `verify-ca`, or `verify-full`, it may
```

Comment:
is it possible / sensible to break out the stories for adding this functionality into an initial opinionated config, and gradually add in options to specify some of the items below? at current it seems very flexible and configurable, but I'm wondering if that will also mean it is harder to implement in one go.

Reply:
Yup. Latest `Stories` section shows the stories each handler should have in agreement with what you wrote